data-processor/src/test/groovy/io/micronaut/data/model/query/builder/SqlQueryBuilderSpec.groovy

```diff
@@ -335,7 +335,7 @@ interface MyRepository {
         def result = encoder.buildInsert(AnnotationMetadata.EMPTY_METADATA, entity)
 
         expect:
```

Comment:
Side effect of fix, by default foreign key uses `_id` suffix and due to the error it used `myId` field name and created `_my_id`. If some users might see issue because of this, then it can be overridden using `@MappedProperty`.